core/src/main/scala/chisel3/Mem.scala

```diff
@@ -571,14 +564,11 @@ sealed class SyncReadMem[T <: Data] private[chisel3] (
     implicit evidence: T <:< Vec[_]
   ): T = {
     implicit val sourceInfo = UnlocatableSourceInfo
```

Comment:
Can you also just make the sourceInfo an implicit argument rather than using `UnlocatableSourceInfo`?